### ISSUE TITLE
Screen reader not working properly for text input

### ISSUE THREAD

**Issue by jere-sudo** [electron/electron#45920]
### Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

35.0.0

### What operating system(s) are you using?

Windows

### Operating System Version

Microsoft Windows 11 Enterprise 10.0.26100 N/A Build 26100

### What arch are you using?

x64

### Expected Behavior

When inserting or deleting a from the following input field <input type="text" />, the narrator should announce the inserted or deleted character.

### Actual Behavior

The screen will announce the whole value when inserting or deleting a character. E.g. say the original value in the field was 12345 and we insert 6 to the end of the input, the screen reader will announce "edit 123456"

### Testcase Gist URL

https://gist.github.com/jere-sudo/86cf3b08a371b557fe8e31da8dc20715

### Additional Information

I'm using the built in windows narrator. It can be easily activated with "Windows logo key + Ctrl + Enter". The accessibility behavior works as intended on chrome and edge browsers. You can test it by opening the index.html file on chrome or edge.

**Comment by nikwen** [comment]
Thanks for reporting! Do you know if it is a new issue in Electron 35 or if the behavior was the same on Electron 34?

**Comment by mahmed890** [comment]
-

**Comment by jere-sudo** [comment]
> Thanks for reporting! Do you know if it is a new issue in Electron 35 or if the behavior was the same on Electron 34?

Quickly tested the with electron 34 and the behavior appears there. In our project, we've been using an older electron 25.9.3 where this behavior occurs as well. So I assume this behavior has been present for a while.

**Comment by Niranjan945** [comment]
Hi, I’m Niranjan from India, and I’m interested in contributing to this project. I’m relatively new to this repository and would like to help resolve this issue. However, as I’m still familiarizing myself with the codebase and the problem, I’d appreciate it if someone could provide more context or clarify what needs to be implemented.

Could you please explain the issue in more detail or point me to any relevant documentation? I’d be happy to assist and learn along the way!

Looking forward to your guidance.
